File: Upcoder/Lập trình cơ bản/7.Other/TINH_TONG/LTCB-4.py
def DequyS(n):
    if n == 1:
        return 0.5
    return DequyS(n - 1) + (1.0 / (2 * n))

n = int(input())
print(round(DequyS(n),3), end = "")  # In kết quả với 3 chữ số thập phân

File: Upcoder/Lập trình cơ bản/7.Other/TINH_TONG/LTCB-4.cpp
#include <bits/stdc++.h>
using namespace std;

double DequyS(int n)
{
    if (n == 1)
        return 1.0/2.0;
    return DequyS(n - 1) + (1.0 / (2 * n));  // Sửa 1 thành 1.0 để đảm bảo chia số thực
}

int main()
{
    int n;
    cin >> n;
    cout << fixed << setprecision(3) << DequyS(n);  // In kết quả với 3 chữ số thập phân
    return 0;
}
